### PR TITLE
Support scala 2.12

### PR DESCRIPTION
**Pull request by melrief**
I had to remove the tests using Stack because Stack is deprecated in scala 2.12

### CODE REVIEW

**Comment by jcazevedo** [comment]
We should also update the Travis config and the README accordingly.

**Comment by jcazevedo** [comment]
We should also update the Travis config and the README accordingly.

**Comment by jcazevedo** [comment]
LGTM

File: src/test/scala/pureconfig/PureconfSuite.scala [hunk 1]
```diff
@@ -373,12 +373,6 @@ class PureconfSuite extends FlatSpec with Matchers with OptionValues with TryVal
     saveAndLoadIsIdentity(ConfWithQueueOfFoo(Queue(Foo(1), Foo(2))))
   }
 
-  case class ConfWithStackOfFoo(stack: collection.mutable.Stack[Foo])
```
Comment (leifwickland) [inline src/test/scala/pureconfig/PureconfSuite.scala:376]: To preserve a test on a mutable data structure, we could switch to the mutable queue.

```
  case class ConfWithMutableQueueOfFoo(stack: collection.mutable.Queue[Foo])

  it should s"be able to save and load configurations containing mutable.Queue" in {
    saveAndLoadIsIdentity(ConfWithMutableQueueOfFoo(collection.mutable.Queue(Foo(1))))
  }
```

File: README.md [hunk 2]
```diff
@@ -48,13 +48,13 @@ uses the raw configurations to do its magic.
 
 In the sbt configuration file:
 
-use scala `2.10` or `2.11`:
+use scala `2.10`, `2.11` or `2.12`:
 
 ```scala
 scalaVersion := "2.11.8" // or "2.10.5"
```
Comment (leifwickland) [inline README.md:54]: Nitpick: 2.12 could appear on this line too.

Alternatively, I think you could probably just drop this section, as it seems:
- A given that a `scalaVersion` need to be specified.
- overspecific insofar as pureconfig doesn't have a dependency on a specific patch version of scala.

**Review by leifwickland — COMMENTED** [review]
LGTM 👍 
Thanks for doing this!

**Pull request merged**